### PR TITLE
500 system tests tlake

### PR DESCRIPTION
**Pull request by tlake**
# What does this pull request do?
Updates system tests to pass with new `api-refactor` changeset. Tests have largely been updated to make use of the stateless deployment pattern.

# How should this be tested?
Checkout this branch, then
```
cd layer0/tests/system/
make test
```
Don't run `make test-all` - that will kick off the Windows system test. Since we intend to (at least temporarily) remove Windows support from v0.11.x in a future commit, there's no need to test Windows functionality at this time.

(Also, the Windows test will just fail.)

## Requirements
- Fresh builds of the Layer0 binaries in your `$PATH` (in other words, running `l0 anything` should be using the versions of everything as they exist in the branch currently)
- A Fargate-compatible Layer0 instance. `l0-setup init` with the following considerations:
  - **Private Docker:** No.
  - **Source:** `${local-layer0-dir}//setup/module`
  - **Version:** `tlake500`
  - **AWS Region:** `us-east-1`
- A working terraform plugins configuration. Personally, I've been unsuccessful in setting up my system to work like zpatrick's, but here's the workaround I use:
  - Ensure that the `terraform-provider-layer0` binary is built and available in your `$PATH`.
  - Link to it from each case directory, then run `terraform init` manually:
    - `cd layer0/tests/system/cases/`
    - `for d in datasources/ dead_service_recreated/ environment_link/ import/ service_scale/ windows_service/ ; do cd $d ; ln -s $(which terraform-provider-layer0) . ; terraform init ; cd - ; done`

# Checklist
- ~[ ] Unit tests~
- ~[ ] Smoke tests (if applicable)~
- [x] System tests (if applicable)
- ~[ ] Documentation (if applicable)~

This is part of the work for #500.

### CODE REVIEW

File: tests/system/system_test.go [hunk 1]
```diff
@@ -34,11 +34,12 @@ func NewSystemTest(t *testing.T, dir string, vars map[string]string) *SystemTest
 	layer0 := clients.NewLayer0TestClient(t, vars["endpoint"], vars["token"])
 
 	// download modules using terraform init
-	stdoutStderr, err := tfContext.Init()
-	if err != nil {
-		t.Fatalf("%s", err.Error())
-	}
-	log.Printf("%s\n", stdoutStderr)
+	tfContext.Init()
+	// stdoutStderr, err := tfContext.Init()
```
Comment (diemonster) [inline tests/system/system_test.go]: is this intentional?

File: tests/system/cases/datasources/outputs.tf [hunk 2]
```diff
@@ -145,3 +181,11 @@ output "service_scale" {
 output "service_scale_expected" {
   value = "${layer0_service.datasources.scale}"
 }
+
+# output "service_stateful" {
```
Comment (diemonster) [inline tests/system/cases/datasources/outputs.tf]: is this intended to be commented out or removed?

File: tests/system/cases/service_scale/main.tf [hunk 3]
```diff
@@ -6,9 +6,11 @@ provider "layer0" {
 
 resource "layer0_environment" "ss" {
   name = "ss"
+  scale = 3
```
Comment (sesh-kebab) [inline tests/system/cases/service_scale/main.tf]: Scaling the environment to 3 doesn't seem necessary?

Comment (tlake) [reply tests/system/cases/service_scale/main.tf]: Correct! That's a typo. :)

✅

File: tests/system/cases/modules/sts/main.tf [hunk 4]
```diff
@@ -1,7 +1,8 @@
 resource "layer0_load_balancer" "sts" {
-  name        = "sts"
+  name        = "${var.name}"
   environment = "${var.environment_id}"
   private     = "${var.private}"
+  type        = "${var.stateful ? "classic" : "application"}"
```
[the remark this answers is not in the document]
Comment (tlake) [reply tests/system/cases/modules/sts/main.tf]: 👌

**Comment by tlake** [comment]
With @diemonster's blessing, I'm merging this so we can get to a state where we can shelve the `api-refactor` work and prevent it from blocking other work.

**Pull request merged**